safenode/src/network/event.rs

```diff
@@ -89,40 +86,24 @@ impl NetworkSwarmLoop {
             SwarmEvent::Behaviour(NodeEvent::Kademlia(event)) => match event {
                 KademliaEvent::OutboundQueryProgressed {
                     id,
-                    result: QueryResult::StartProviding(_),
-                    ..
-                } => {
-                    let sender: oneshot::Sender<Result<()>> = self
-                        .pending_start_providing
-                        .remove(&id)
-                        .ok_or(Error::Other(
-                            "Completed query to be previously pending.".to_string(),
-                        ))?;
-                    let _ = sender.send(Ok(()));
-                }
-                KademliaEvent::OutboundQueryProgressed {
-                    id,
-                    result:
-                        QueryResult::GetProviders(Ok(GetProvidersOk::FoundProviders {
-                            providers, ..
-                        })),
+                    result: QueryResult::GetClosestPeers(Ok(closest_peers)),
                     ..
                 } => {
-                    if let Some(sender) = self.pending_get_providers.remove(&id) {
+                    if let Some(sender) = self.pending_get_closest_nodes.remove(&id) {
                         sender
-                            .send(providers)
+                            .send(closest_peers.peers.into_iter().collect())
                             .map_err(|_| Error::Other("Receiver not to be dropped".to_string()))?;
-
-                        // Finish the query. We are only interested in the first result.
-                        self.swarm
-                            .behaviour_mut()
-                            .kademlia
-                            .query_mut(&id)
-                            .ok_or(Error::Other("Query should exist".to_string()))?
-                            .finish();
                     }
                 }
-                _ => {}
+                KademliaEvent::RoutingUpdated { .. } => {
+                    self.event_sender.send(NetworkEvent::PeerAdded).await?;
+                }
+                KademliaEvent::InboundRequest { request } => {
+                    info!("got inboumd request: {request:?}");
+                }
+                todo => {
+                    panic!("KademliaEvent has not been implemented: {todo:?}");
```

Comment:
I wouldn't panic but log a warning, I think we may get an event and we don't want our node to crash just because we are not interested in such an event or because we still didn't handle it.